### PR TITLE
fix: allow interception for all requests

### PR DESCRIPTION
**Pull request by Lightning00Blade**

### CODE REVIEW

File: src/bidiMapper/domains/network/NetworkStorage.ts [hunk 1]
```diff
@@ -176,6 +186,10 @@ export class NetworkStorage {
     pathname,
     search,
   }: Network.UrlPatternPattern): string {
+    if (!protocol && !hostname && !port && !pathname && !search) {
```
Comment (thiagowfx) [inline src/bidiMapper/domains/network/NetworkStorage.ts:189]: Augment the unit test for this case

File: src/bidiMapper/domains/network/NetworkStorage.ts [hunk 2]
```diff
@@ -66,7 +66,7 @@ export class NetworkStorage {
    */
   addIntercept(value: {
     urlPatterns: Network.UrlPattern[];
-    phases: Network.InterceptPhase[];
+    phases: [Network.InterceptPhase, ...Network.InterceptPhase[]];
```
Comment (thiagowfx) [inline src/bidiMapper/domains/network/NetworkStorage.ts:69]: Does this mean an array with at least one element?

File: src/bidiMapper/domains/network/NetworkProcessor.ts [hunk 3]
```diff
@@ -68,7 +68,11 @@ export class NetworkProcessor {
 
     const intercept: Network.Intercept = this.#networkStorage.addIntercept({
       urlPatterns: parsedUrlPatterns,
-      phases: params.phases,
+      // We checked that phases has at least one entry
```
Comment (thiagowfx) [inline src/bidiMapper/domains/network/NetworkProcessor.ts:71]: I don't think this is needed. There's a `params.phases.length === 0` exception right at the beginning. Revert this chunk.

File: src/bidiMapper/domains/network/NetworkProcessor.ts [hunk 4]
```diff
@@ -334,6 +338,17 @@ export class NetworkProcessor {
           return urlPattern;
         }
         case 'pattern':
+          // No params signifies intercept all
+          if (
```
Comment (thiagowfx) [inline src/bidiMapper/domains/network/NetworkProcessor.ts:342]: Is this chunk really needed, given the change to return `'*'` in `buildUrlPatternString`?

File: src/bidiMapper/domains/network/NetworkProcessor.ts [hunk 5]
```diff
@@ -367,6 +382,8 @@ export class NetworkProcessor {
           }
 
           try {
+            // TODO: change this as if we have a single param form Pattern
```
Comment (thiagowfx) [inline src/bidiMapper/domains/network/NetworkProcessor.ts:385]: nit: it's not clear to me what the AI is here. Could you rephrase?

File: src/bidiMapper/domains/network/NetworkStorage.ts [hunk 6]
```diff
@@ -111,13 +111,23 @@ export class NetworkStorage {
     const patterns: Protocol.Fetch.RequestPattern[] = [];
 
     for (const value of this.#interceptMap.values()) {
-      for (const urlPatternSpec of value.urlPatterns) {
-        const urlPattern: string =
-          NetworkStorage.cdpFromSpecUrlPattern(urlPatternSpec);
-        for (const phase of value.phases) {
+      for (const phase of value.phases) {
```
Comment (thiagowfx) [inline src/bidiMapper/domains/network/NetworkStorage.ts:114]: Why do we introduce this new loop?

File: src/bidiMapper/domains/network/NetworkStorage.ts (hunk 2 above)
Comment (Lightning00Blade) [reply src/bidiMapper/domains/network/NetworkStorage.ts:69]: Yes, the first one will always be present. See https://github.com/google/cddlconv/issues/16 for cddlconv issue.

File: src/bidiMapper/domains/network/NetworkProcessor.ts (hunk 3 above)
Comment (Lightning00Blade) [reply src/bidiMapper/domains/network/NetworkProcessor.ts:71]: TS is not smart enough to understand this sadly.

File: src/bidiMapper/domains/network/NetworkProcessor.ts (hunk 5 above)
Comment (Lightning00Blade) [reply src/bidiMapper/domains/network/NetworkProcessor.ts:385]: It was for me, as I was planning to do more things in the parsing.

File: src/bidiMapper/domains/network/NetworkStorage.ts (hunk 6 above)
Comment (Lightning00Blade) [reply src/bidiMapper/domains/network/NetworkStorage.ts:114]: We did not we switched the order of the loops as `Phase` is always at least one and `UrlPatterns` may be empty.

**Comment by thiagowfx** [comment]
When you mark it as ready for review I'll take another look

**Review by thiagowfx — APPROVED** [review]

**Comment by thiagowfx** [comment]
I expect CI to fail. Only 2 expectations were updated, it should have been 3.

**Comment by Lightning00Blade** [comment]
I remember running `./tools/runWPTall.sh`. Maybe there was some issue let me try again.

**Comment by Lightning00Blade** [comment]
@thiagowfx I can't seem to run Mapper with ChromeDriver locally, can you check if you can on this PR?

**Comment by thiagowfx** [comment]
Ran `tools/runWptAll.sh` and pushed.

File: wpt-metadata/mapper/headless/webdriver/tests/bidi/network/add_intercept/invalid.py.ini
Comment (thiagowfx) [inline wpt-metadata/mapper/headless/webdriver/tests/bidi/network/add_intercept/invalid.py.ini:1]: These `invalid.py` additions are regressions

**Review by thiagowfx — CHANGES_REQUESTED** [review]
Look into invalid.py regressions

**Review by thiagowfx — APPROVED** [review]

**Comment by thiagowfx** [comment]
I cannot easily see what has changed since the last commit but, assuming the tests pass, LGTM
